### PR TITLE
Rename EvosMovesPointerTable to EvosAttacksPointers to sync with pokecrystal #315

### PR DESCRIPTION
**Pull request by KqesaR**
This PR have for goal to sync the name of pointer for evo attacks like pokecrystal.

Related to the issue #315

### CODE REVIEW

**Comment by Rangi42** [comment]
Good idea, thanks! `evos_moves.asm` should also be renamed to `evos_attacks.asm`.

**Comment by KqesaR** [comment]
I have updated my PR to take in consideration your suggestion @Rangi42 :)

File: data/pokemon/evos_attacks.asm [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ; See constants/pokemon_data_constants.asm
 ; The max number of evolutions per monster is MAX_EVOLUTIONS
 
-EvosMovesPointerTable:
-	table_width 2, EvosMovesPointerTable
+EvosAttacksPointers:
+	table_width 2, EvosAttacksPointers
 	dw RhydonEvosMoves
```
Comment (dannye) [inline data/pokemon/evos_attacks.asm:6]: The problem now is that the rest of this file continues to use "EvosMoves" (`RhydonEvosMoves` etc).

File: engine/menus/party_menu.asm [hunk 2]
```diff
@@ -122,15 +122,15 @@ RedrawPartyMenu_::
 	ld c, a
 	add hl, bc
 	ld de, wEvosMoves
```
Comment (dannye) [inline engine/menus/party_menu.asm:124]: and `wEvosMoves` as well

**Comment by dannye** [comment]
Sorry for being contrarian (again), but I argue that pokecrystal should be updated to match pokered in this case.

Both repos already primarily use "move" (data/moves/ etc) over "attack" to refer to pokemon moves. The mainline games do as well (move relearner, move deleter, etc).
Plus, "attack" is a stat like defense etc. so better to not let that word get confused with moves.

**Comment by KqesaR** [comment]
I can update pokecrystal instead of pokered if you prefer

My wish is to contribute according to your recommendations, it is not a concern to rectify my PRs

**Comment by dannye** [comment]
We can wait for @Rangi42 to chime in too, in case she feels strongly one way or the other.

**Comment by Rangi42** [comment]
> Sorry for being contrarian (again), but I argue that pokecrystal should be updated to match pokered in this case.
> 
> Both repos already primarily use "move" (data/moves/ etc) over "attack" to refer to pokemon moves. The mainline games do as well (move relearner, move deleter, etc). Plus, "attack" is a stat like defense etc. so better to not let that word get confused with moves.

This is a good point. One thing is that we have `NUM_MOVES EQU 4` (how many a mon can learn) and `NUM_ATTACKS EQU (the number of different moves)`. (That's the case in the Gen 1 and 2 repos.) It would probably make sense for those to be renamed `NUM_MON_MOVES` and `NUM_MOVES`, but I wonder if changing the meaning of `NUM_MOVES` would be too confusing.

**Comment by dannye** [comment]
> This is a good point. One thing is that we have `NUM_MOVES EQU 4` (how many a mon can learn) and `NUM_ATTACKS EQU (the number of different moves)`. (That's the case in the Gen 1 and 2 repos.) It would probably make sense for those to be renamed `NUM_MON_MOVES` and `NUM_MOVES`, but I wonder if changing the meaning of `NUM_MOVES` would be too confusing.

Hmm good point also. Swapping out those names might cause some confusion but probably worth it imo. Those suggested names are much better.

**Comment by iimarckus** [comment]
Reusing an old name is gratuitously unfriendly and discourages people from updating their repos.

**Comment by dannye** [comment]
> Reusing an old name is gratuitously unfriendly and discourages people from updating their repos.

Yeah that's fair. I think `NUM_MOVES` can be renamed to `NUM_MON_MOVES` at the very least because that's just all around better.
Maybe `NUM_ATTACKS` should just stay the way it is. It isn't *horrible* after all.

**Comment by Rangi42** [comment]
@KqesaR Thanks for getting this synchronization started. I'm closing this PR since we decided to go the other way and use "moves" instead of "attacks" in all four repos (except for keeping `NUM_ATTACKS` and renaming `DEF NUM_MOVES EQU 4` to `DEF NUM_MON_MOVES EQU 4`, so there just won't be a "`NUM_MOVES`" constant).

**Pull request closed without merging**